### PR TITLE
AIs can no longer strip people

### LINKED ISSUE #4950: AI can strip people
## Round ID:
31642

## Testmerges:
#4939: 'Hearing revert round whatever [DNM]' by francinum at commit db6b8402c0
#4836: 'Stat panel verb caching' by PowerfulBacon at commit ba7de817fa

## Reproduction:
Be cuffed/clothed
Stand next to AI
AI can uncuff/strip people like humans

### PR DESCRIPTION
**Pull request by imsynthex**
## About The Pull Request

This might be an awful way of doing it, but that's the way it's done with larvae, and it works

I think this was meant to be fixed before in https://github.com/BeeStation/BeeStation-Hornet/pull/4590 but that just didn't work

## Why It's Good For The Game

Closes https://github.com/BeeStation/BeeStation-Hornet/issues/4950

## Changelog
:cl:
fix: AIs can no longer strip people
/:cl:

### CODE REVIEW

**Comment by St0rmC4st3r** [comment]
*snap
*snap

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 1]
```diff
@@ -1100,5 +1100,13 @@
 	if(.)
 		end_multicam()
 
+/mob/living/silicon/ai/stripPanelUnequip(obj/item/what, mob/who)
+	to_chat(src, "<span class='warning'>You don't have any arms!</span>")
```
Comment (Archanial) [inline code/modules/mob/living/silicon/ai/ai.dm]:
```suggestion
```
It doesn't need any message

**Review by Archanial — CHANGES_REQUESTED** [review]

**Comment by Archanial** [comment]
The actual solution to the issue lays in mob.dm.
The best way would be to just not show up the panel to something like AI which should never be able to strip people.
https://github.com/BeeStation/BeeStation-Hornet/blob/2d4937b60ca8546bb87fbb2edae5aa96e1bcfd87/code/modules/mob/mob.dm#L805
this is the check for this.
All you need to add is ` && !isAI(U)`.

**Comment by imsynthex** [comment]
yeah, i was looking for that, thanks, i'll update this in a few hours to use mob.dm

also maybe change larvae over to that 😳

**Review by ike709 — DISMISSED** [review]

File: code/modules/mob/mob.dm [hunk 2]
```diff
@@ -802,7 +802,7 @@
 		var/mob/M = dropping
 		if(ismob(user))
 			var/mob/U = user
-			if(!iscyborg(U) || U.a_intent == INTENT_HARM)
+			if(!iscyborg(U) && !isAI(U) || U.a_intent == INTENT_HARM)
```
Comment (Archanial) [inline code/modules/mob/mob.dm]:
```suggestion
			if((!iscyborg(U) || U.a_intent == INTENT_HARM) && !isAI(U))
```
AIs shouldn't have strip panel even on harm and besides both AI and borg are the same subtype.

**Review by Archanial — CHANGES_REQUESTED** [review]

**Review by ike709 — APPROVED** [review]

**Review by Archanial — APPROVED** [review]
:+1:

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**